tests/test_bash_done_spam.py

```
@@ -0,0 +1,255 @@
+class InteractiveBash:
+    def __init__(self, bash_path: str, env: dict[str, str]) -> None:
+        self.bash_path = bash_path
+        self.env = env
+        self.pid: int | None = None
+        self.fd: int | None = None
+        self.output = bytearray()
+
+    def __enter__(self) -> "InteractiveBash":
+        pid, fd = pty.fork()
+        if pid == 0:
+            os.execvpe(
```

Ruff (0.15.14)

[error] Starting a process without a shell

(S606)